### PR TITLE
Use findings and alerts models, dtos from common utils

### PR DESCRIPTION
**Pull request by eirsep**
Signed-off-by: Surya Sashank Nistala <snistala@amazon.com>

### CODE REVIEW

File: alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetAlertsAction.kt [hunk 1]
```diff
@@ -105,37 +117,78 @@ class TransportGetAlertsAction @Inject constructor(
...
         }
     }
 
-    fun resolve(
+    /** Precedence order for resolving alert index to be queried:
+     1. alertIndex param.
+     2. alert index mentioned in monitor data sources.
+     3. Default alert indices pattern
+     */
+    suspend fun resolveAlertsIndexName(getAlertsRequest: GetAlertsRequest): String {
```
Comment (getsaurabh02) [inline alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetAlertsAction.kt:141]: Lets add a unit test to verify name resolution

File: alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetFindingsAction.kt [hunk 2]
```diff
@@ -122,19 +127,48 @@ class TransportGetFindingsSearchAction @Inject constructor(
...
+                    actionListener.onFailure(t)
                 } catch (t: Exception) {
                     actionListener.onFailure(AlertingException.wrap(t))
                 }
             }
         }
     }
 
-    suspend fun search(searchSourceBuilder: SearchSourceBuilder): GetFindingsResponse {
+    suspend fun resolveFindingsIndexName(findingsRequest: GetFindingsRequest): String {
```
Comment (getsaurabh02) [inline alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetFindingsAction.kt:142]: same as above - we should verify the name resolution logic via unit test.

**Review by getsaurabh02 — COMMENTED** [review]
Thanks @eirsep . LGTM, couple of suggestions around testing.

File: alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetAlertsAction.kt (hunk 1 above)
Comment (eirsep) [reply alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetAlertsAction.kt:141]: this has been verified in the integ test in the following scenarios:

Verify get alerts for monitor WITHOUT custom alerts index.
Verify get alerts for monitor WITH custom alerts index and request has monitor ID.

File: alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetFindingsAction.kt (hunk 2 above)
Comment (eirsep) [reply alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportGetFindingsAction.kt:142]: same test coverage as detailed for get alerts.

**Review by getsaurabh02 — APPROVED** [review]

**Review by sbcd90 — APPROVED** [review]

**Pull request merged**